scripts/application_sync_providers.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import logging
+import sys
+import pprint
+
+import application_to_provider
+import django;
+
+django.setup()
+import core.models
+
+description = """
+This script performs a one-way synchronization of Applications (a.k.a. images)
+from a master Provider to one or more replica Providers. Only synchronizes
+active (i.e. non-end-dated) Applications and ApplicationVersions.
+
+For each Application known to Atmosphere(2), if there is at least one
+ApplicationVersion with a ProviderMachine + InstanceSource on the master
+Provider, then that Application is considered for replication.
+
+For each replica Provider given, if the replica Provider does not have a
+ProviderMachine + InstanceSource for every ApplicationVersion of an
+Application considered for replication, then application_to_provider is run
+to migrate that Application to the replica Provider.
+
+If --irods-conn and --irods-collections are defined, then iRODS transfer mode
+will be used for each provider ID defined in the --irods-collections JSON.
+Pure Glance API will be used to transfer image data for any providers not
+defined in --irods-collections. When using iRODS Transfer, the same
+requirements and caveats apply as detailed in application_to_provider.py.
+
+NOTE that --limit-app-ids cannot be the last argument before the positional
+arguments; otherwise argparse will get confused and complain about "too few
+arguments"
+"""
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description=description, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument("master_provider_id", type=int, help="Master provider ID")
+    parser.add_argument("replica_provider_id", type=int, nargs='+', help="Replica provider ID(s)")
+    # https://stackoverflow.com/questions/18608812/accepting-a-dictionary-as-an-argument-with-argparse-and-python
+    parser.add_argument("--glance-client-versions", type=json.loads,
+                        metavar='{\"1\": 2, \"2\": 1}',
+                        help="JSON specifying Glance client version for each provider ID")
```

Comment:
Maybe. I found that that legacy (Havana) providers require Glance API V1 client in order to create an image with a specific UUID. Do we have that kind of information stored somewhere?